clients/ui/frontend/src/app/context/modelCatalog/ModelCatalogContext.tsx

```diff
@@ -93,6 +103,26 @@ export const ModelCatalogContextProvider: React.FC<ModelCatalogContextProviderPr
   const [selectedSourceLabel, setSelectedSourceLabel] = React.useState<
     ModelCatalogContextType['selectedSourceLabel']
   >(CategoryName.allModels);
+  const [performanceViewEnabled, setPerformanceViewEnabled] = React.useState(false);
+  const [performanceFiltersChangedOnDetailsPage, setPerformanceFiltersChangedOnDetailsPage] =
+    React.useState(false);
+
+  const location = useLocation();
+  const prevLocationRef = React.useRef(location.pathname);
+
+  React.useEffect(() => {
+    const prevPath = prevLocationRef.current;
+    const currentPath = location.pathname;
+
+    const wasOnDetailsPath = isModelDetailsPage(prevPath);
+    const isNowOnDetailsPath = isModelDetailsPage(currentPath);
+
+    if (!wasOnDetailsPath && isNowOnDetailsPath) {
+      setPerformanceFiltersChangedOnDetailsPage(false);
+    }
+
+    prevLocationRef.current = currentPath;
+  }, [location.pathname]);
```

Comment:
I don't think we need the prevLocationRef or the isModelDetailsPage util, we can simply have an effect in PerformanceInsightsView that runs once on mount to reset this state. Will comment below.